src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/PackageSourceTelemetry.cs

```diff
@@ -5,130 +5,188 @@
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 using NuGet.Common;
 using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
 using NuGet.Protocol.Utility;
 
 namespace NuGet.VisualStudio.Telemetry
 {
     public sealed class PackageSourceTelemetry : IDisposable
     {
         private readonly ConcurrentDictionary<string, Data> _data;
-        private readonly IDictionary<string, PackageSource> _sources;
+        private readonly IDictionary<string, SourceRepository> _sources;
```

Comment:
Why did you need to use a SourceRepository?
I see you are using PackageSource anyways.

Reply:
Because `PackageSource.ProtocolVersion` is useless, whereas `SourceRepository.GetFeedType` is accurate